debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/HostUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Host string parsing utilities
+ */
+public final class HostUtils {
+
+    private HostUtils() {
+        // intentionally private;
+    }
+
+    /**
+     * Regular expression that extracts the hosts for the replica sets. The raw expression is
+     * {@code ((([^=]+)[=])?(([^/]+)\/))?(.+)}.
+     */
+    private static final Pattern HOST_PATTERN = Pattern.compile("((([^=]+)[=])?(([^/]+)\\/))?(.+)");
+
+    public static ReplicaSet parse(String hosts) {
```

Comment:
Is this intended to fully parse all forms of [mongodb.hosts](https://debezium.io/documentation/reference/stable/connectors/mongodb.html#mongodb-property-mongodb-hosts)](https://debezium.io/documentation/reference/stable/connectors/mongodb.html#mongodb-property-mongodb-hosts)?  From the description, it looks as though it supports multiple hosts with ports although it's unclear how that looks.

Reply:
In terms of cohesion, this looks as though it could go in `MongoUtil`. In particular, `MongoUtil#parseAddresses` looks very similar.

Reply:
`mongodb.hosts` unfortunately allows for some pretty invalid configurations... it's essetnially a `;` delimitted list of replica sets in a form of `<shardName>=<rsName>/<addr>:<port>,<addr2>:<port2>,...` where addres and port, shard name is optional, and the name is sort of optional. There is nothign mandating this to be related replica sets or what kind or replica sets this ought to be when connecting to a sharded cluster (hence all those issues met by users when actually trying to use this property instead of the `connection.string` introduced with 2.0).

Reply:
The idea is to deprecate the hots in time -- we will have to figure out the degree of compatibility first though. Parsing a certainly correct connection string is not possible due to unclear semantics of `mongodb.hosts`. Probably the easiest thing would be to just take the addresses and hosts and `mongodb://` protocol.

Reply:
Thanks for clarifying. Is the plan to deprecate this property for removal in 3.0?